apis/eks/v1alpha1/types.go

```diff
@@ -0,0 +1,383 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// AMITypes is a type of AMI.
+type AMITypes string
+
+// AMI types available for NodeGroup.
+const (
+	AMITypesAl2X8664    AMITypes = "AL2_x86_64"
+	AMITypesAl2X8664Gpu AMITypes = "AL2_x86_64_GPU"
+)
+
+// NodeGroupIssueCode is a code for an issue in a NodeGroup.
+type NodeGroupIssueCode string
+
+// Types of NodeGroup issue codes.
+const (
```

Comment:
I think having the set of possible inputs or outputs is useful when the set does not change an we get a validation that'll improve the UX. While `NodeGroupIssueCode` values probably won't change, I can't see how useful it is to have all possibilities in the codebase. So, I'd prefer removing them and just having `string.`

Reply:
Sounds reasonable to me